scripts/convert_csv_to_json.py

```diff
@@ -10,43 +10,47 @@ def generate_json_files():
     source_root = './source-data'
     output_root = './data'
 
+    for source_region_directory_name in os.listdir(source_root):
+        source_region_directory = os.path.join(source_root, source_region_directory_name)
 
-    for source_directory_name in os.listdir(source_root):
+        for language_directory_name in os.listdir(source_region_directory):
+            source_directory = os.path.join(source_region_directory, language_directory_name)
 
-        source_directory = os.path.join(source_root, source_directory_name)
+            for source_file_name in os.listdir(source_directory):
 
-        for source_file_name in os.listdir(source_directory):
+                source_file_location = os.path.join(source_directory, source_file_name)
 
-            source_file_location = os.path.join(source_directory, source_file_name)
+                if os.path.isfile(source_file_location):
 
-            if os.path.isfile(source_file_location):
+                    source_file = open(source_file_location, 'r')
 
-                source_file = open(source_file_location, 'r')
+                    output_file_name = source_file_name.replace('.csv', '.json')
+                    output_file_location = os.path.join(output_root,
+                                                        source_region_directory_name,
+                                                        language_directory_name,
+                                                        output_file_name)
 
-                output_file_name = source_file_name.replace('.csv', '.json')
-                output_file_location = os.path.join(output_root, source_directory_name, output_file_name)
+                    json_data = []
 
-                json_data = []
+                    with open(source_file_location, newline='') as csv_file:
+                        for row in csv.DictReader(csv_file, fieldnames=['term']):
+                            json_data.append(row.get('term'))
+                    try:
+                        with open(output_file_location, 'w', encoding='utf8') as f:
+                            json.dump(json_data, f, indent=3, ensure_ascii=False)
+                            f.write("\n")
+                    except FileNotFoundError:
+                        print(
+                            f"Output could not be written to {output_file_location}\nDoes the output folder"
+                            f"{output_root}/{source_region_directory_name}/{language_directory_name} exist?"
+                        )
+                        return
```

Comment:
Can we get rid of this try/except and just create the directory when it doesn't exist?. I don't see any issues with doing that and avoid me having to create the directory myself.
```suggestion
                    os.makedirs(
                        os.path.dirname(
                            f"{output_root}/{source_region_directory_name}/{language_directory_name}"
                        ),
                        exist_ok=True,
                    )
                    with open(output_file_location, 'w+', encoding='utf8') as f:
                        json.dump(json_data, f, indent=3, ensure_ascii=False)
                        f.write("\n")
```

Reply:
In the original implementation of this I deliberately didn't create output folders on the fly as I was using the language codes provided by the business (within the file headers) as the output location. This was a useful way of validating they had provided the correct structure and values in their files (as these were inconsistent). Agreed that if we had more arbitrary values and permutations of possible folders this would be the desired approach, but I'm bot sure its needed for the current useage.

Reply:
However, this now relies on own our structure hence there should be no issue creating it on the fly?. The structure doesn't come from the business any more, we defined our own structure for source data which matches what we want for output data. Please correct me if I have misunderstood.

Reply:
You haven't misunderstood. However, there are no more regions or language permutations we need to support given the requirements. If that changes and it causes an effort/risk over and above simply changing this script to create them on the fly we can do so at that time as part of that requirement.

Reply:
My comment was regarding creating the output directory automatically?. Doing so would avoid human error, as long as source dir structure is correct?. I don't quite understand what risk we are talking about. I don't mean auto create the permutations we need for the source directory, I am talking just about the output directory.